base/db/tests/testthat/test.insert.R

```diff
@@ -0,0 +1,53 @@
+library(PEcAn.DB)
+library(DBI)
+library(testthat)
+library(dplyr)
+context("SQL insertion helper functions")
+
+if (requireNamespace("RSQLite")) {
+  library(RSQLite)
+  iris_file <- tempfile(fileext = ".sqlite")
+  irisdb <- dbConnect(SQLite(), iris_file)
+  iris <- as_tibble(iris) %>%
+    mutate(
+      Species = as.character(Species)
+    )
+
+  copy_to(irisdb, iris[1,], "iris", overwrite = TRUE)
+  # Add extra column to see if it's successfully ignored
+  iris2 <- mutate(iris, extracol = row_number())
+  iris_insert <- iris2[2:10,]
+  .insert <- insert_table(iris_insert, "iris", irisdb)
+  iris_insert_test <- tbl(irisdb, "iris") %>% collect()
+  test_that(
```

Comment:
In general, yes, I agree. Here, this is a bit fuzzy because the tests build on each other (i.e. the merge only makes sense if the table has already been previously inserted), but I can certainly slide a few of the lines into the `test_that` blocks.